analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.PatternMatching.cs

```diff
@@ -301,6 +318,23 @@ public void DeclarationPattern_Var_Deconstruction_DoesNotSetConstraint()
                 .And.ContainSingle(x => x.HasConstraint(TestConstraint.First) && x.HasConstraint(ObjectConstraint.NotNull));
         }
 
+        [TestMethod]
+        public void DeclarationPattern_ExistingConstraint_DoesNothing()
+        {
+            const string code = @"
+object value = arg switch
+{
+    null => null,
+    Exception ex when Condition => null, // Should not create arg=Null
+    _ => Tag(""Arg"", arg)
+};
+
+static object Tag(string name, object value) => null;";
```

Comment:
Same here

Reply:
The other returns `void` and doesn't play nicely with the `switch`